Grbl_Esp32/src/SDCard.cpp

```diff
@@ -95,33 +98,112 @@ bool SDCard::closeFile() {
     return true;
 }
 
+
 /*
   read a line from the SD card
   strip whitespace
   strip comments per http://linuxcnc.org/docs/ja/html/gcode/overview.html#gcode:comments
   make uppercase
   return true if a line is
 */
-bool SDCard::readFileLine(char* line, int maxlen) {
+SDCard::ReadResult_t SDCard::readFileLine(char* line, int maxlen) {
+
+    // _readyNext is, by convention and program logic, FALSE at this point
+
+    #define DEBUG_READFILELINE 0
+
+    #if DEBUG_READFILELINE
+        debug_serial("readFileLine started: file(%s) size(%d) pos(%d) avail(%d)",
+            _pImpl->_file?"OPEN":"NULL",
+            _pImpl->_file?_pImpl->_file.size():-237,
+            _pImpl->_file?_pImpl->_file.position():-237,
+            _pImpl->_file?_pImpl->_file.available():-237);
+    #endif
+
     if (!_pImpl->_file) {
+        #if DEBUG_READFILELINE
+            debug_serial("readFileLine: returning false because somebody incorrectly called it with a closed file");
+                // fwiw, once a file is opened the operator bool() will NEVER return false until the file is closed,
```

Comment:
Please restrict the use of all caps to acronyms.  The extra emphasis by writing NEVER is unnecessary and jarring.  Similarly HOWEVER in a comment below.  Also the use of interjectory phrases like "fwiw", "in reality", "and burn it", and "By rights" contributes little to the clarity of the comment.